### PR TITLE
Fix(extra7141): Error handling and include missing policy

### PR DESCRIPTION
**Pull request by lazize**
### Context 

Missing permission was exposing AccessDenied error message.  
Also missing check was generating false positive.

### Description

Add check to validate access denied when get document from SSM.
Add missing action permission to allow `ssm:GetDocument`.

Policy actions sorted to keep them in alphabetical order.

Issue #1016

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lazize** [comment]
```shell
$ ./prowler -p default -f sa-east-1 -c extra7141
                          _
  _ __  _ __ _____      _| | ___ _ __
 | '_ \| '__/ _ \ \ /\ / / |/ _ \ '__|
 | |_) | | | (_) \ V  V /| |  __/ |
 | .__/|_|  \___/ \_/\_/ |_|\___|_|v2.7.0-24January2022
 |_| the handy cloud security tool

 Date: Wed Feb  2 22:05:13 -03 2022

 Color code for results:
 -  INFO (Information)
 -  PASS (Recommended value)
 -  WARNING (Ignored by whitelist)
 -  FAIL (Fix required)

 This report is being generated using credentials below:

 AWS-CLI Profile: [default] AWS API Region: [sa-east-1] AWS Filter Region: [sa-east-1]
 AWS Account: [xxxx5262] UserId: [AROAxxxx:xxxxxx]
 Caller Identity ARN: [arn:aws:sts::xxxx5262:assumed-role/xxxxx/xxxxxx]

7.141 [extra7141] Find secrets in SSM Documents - ssm [Critical]
       INFO! sa-east-1: Access Denied trying to get document
```

```shell
$ ./prowler -p default -f sa-east-1 -c extra7141
                          _
  _ __  _ __ _____      _| | ___ _ __
 | '_ \| '__/ _ \ \ /\ / / |/ _ \ '__|
 | |_) | | | (_) \ V  V /| |  __/ |
 | .__/|_|  \___/ \_/\_/ |_|\___|_|v2.7.0-24January2022
 |_| the handy cloud security tool

 Date: Wed Feb  2 22:05:46 -03 2022

 Color code for results:
 -  INFO (Information)
 -  PASS (Recommended value)
 -  WARNING (Ignored by whitelist)
 -  FAIL (Fix required)

 This report is being generated using credentials below:

 AWS-CLI Profile: [default] AWS API Region: [sa-east-1] AWS Filter Region: [sa-east-1]
 AWS Account: [xxxx5262] UserId: [AROAxxxx:xxxxx]
 Caller Identity ARN: [arn:aws:sts::xxxx5262:assumed-role/xxxx/xxxxx]

7.141 [extra7141] Find secrets in SSM Documents - ssm [Critical]
       PASS! sa-east-1: No secrets found in SSM Document test-prowler-leo
```

**Comment by lazize** [comment]
I couldn't make it FAIL. How it check for secret?

My test document is the one below

```json
{
  "schemaVersion": "2.2",
  "description": "Command Document Example JSON Template",
  "parameters": {
    "Message": {
      "type": "String",
      "description": "Example",
      "default": "Hello World"
    },
    "dbpass": {
      "type": "String",
      "description": "Example",
      "default": "temp123"
    }
  },
  "mainSteps": [
    {
      "action": "aws:runPowerShellScript",
      "name": "example",
      "inputs": {
        "runCommand": [
          "Write-Output {{Message}}"
        ]
      }
    }
  ]
}
```

File: util/codebuild/codebuild-prowler-audit-account-cfn.yaml [hunk 1]
```diff
@@ -141,7 +141,7 @@ Resources:
           - id: W28
             reason: "Explicit name is required for this resource to avoid circular dependencies."
     Properties:
-      RoleName: !Sub 'prowler-codebuild-role'
+      RoleName: 'prowler-codebuild-role'
```
Comment (toniblyx) [inline util/codebuild/codebuild-prowler-audit-account-cfn.yaml:144]: why this?

**Review by toniblyx — CHANGES_REQUESTED** [review]
This PR needs to be updated since we already added some new actions.

File: util/codebuild/codebuild-prowler-audit-account-cfn.yaml (hunk 1 above)
Comment (lazize) [reply util/codebuild/codebuild-prowler-audit-account-cfn.yaml:144]: It was failing cfn-lint, as `!Sub` is useless with just static text. There is not variable to perform substitution.

**Review by toniblyx — DISMISSED** [review]
LGTM

File: checks/check_extra7141 [hunk 2]
```diff
@@ -35,11 +35,15 @@ extra7141(){
     if [[ $(echo "$SSM_DOCS" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
         textInfo "$regx: Access Denied trying to list documents" "$regx"
         continue
-    fi  
+    fi
     if [[ $SSM_DOCS ]];then
       for ssmdoc in $SSM_DOCS; do 
         SSM_DOC_FILE="$SECRETS_TEMP_FOLDER/extra7141-$ssmdoc-$regx-content.txt"
-        $AWSCLI $PROFILE_OPT --region $regx ssm get-document --name $ssmdoc --output text --document-format JSON > $SSM_DOC_FILE
+        $AWSCLI $PROFILE_OPT --region $regx ssm get-document --name $ssmdoc --output text --document-format JSON > $SSM_DOC_FILE 2>&1
```
Comment (jfagoagas) [inline checks/check_extra7141]:
```suggestion
        "${AWSCLI}" $PROFILE_OPT --region "${regx ssm}" get-document --name "${ssmdoc}" --output text --document-format JSON > "${SSM_DOC_FILE}" 2>&1
```

File: checks/check_extra7141 [hunk 3]
```diff
@@ -35,11 +35,15 @@ extra7141(){
...
+        if [[ $(grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError' "$SSM_DOC_FILE") ]]; then
```
Comment (jfagoagas) [inline checks/check_extra7141]:
```suggestion
        if [[ $(grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError' "${SSM_DOC_FILE}") ]]; then
```

File: checks/check_extra7141 [hunk 4]
```diff
@@ -35,11 +35,15 @@ extra7141(){
...
+            textInfo "$regx: Access Denied trying to get document" "$regx"
```
Comment (jfagoagas) [inline checks/check_extra7141]:
```suggestion
            textInfo "${regx}: Access Denied trying to get document" "${regx}"
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Thanks @lazize, please check above comments in order to merge this PR.

**Comment by jfagoagas** [comment]
`$PROFILE_OPT` shouldn't be enclosed within "${}" because it contains spaces and we need to extract both values.

**Review by jfagoagas — APPROVED** [review]
Thank you @lazize !!

**Review by toniblyx — APPROVED** [review]

**Pull request merged**